### PR TITLE
Annotations settings: Do not preserve query properties when changing data source

### PR DESCRIPTION
**Pull request by dprokop**
A simple remedy to https://github.com/grafana/support-escalations/issues/8118, basically cleaning all properties from the annotation query, that might have been added to the object. Long term we should implement a datasource change handler similar to the one we use in panel edit so that queries can be imported, or default query can be used, as described in https://github.com/grafana/support-escalations/issues/8118#issuecomment-1786777106

### CODE REVIEW

File: public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx [hunk 1]
```diff
@@ -73,8 +73,15 @@ export const AnnotationSettingsEdit = ({ editIdx, dashboard }: Props) => {
 
   const onDataSourceChange = (ds: DataSourceInstanceSettings) => {
     onUpdate({
-      ...annotation,
```
Comment (torkelo) [inline public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx:76]: what if we change to a different data source of the same type?

**Review by torkelo — APPROVED** [review]
Looks good, except when changing data source of the same type, common to have different Prometheus or influx data sources for different environments. 

could be a bit bad to loose the query then

File: public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx:76]: good point, pushing in a sec

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lgtm for a quick fix 👍

**Pull request merged**